### PR TITLE
Signing mvn deployments

### PR DESCRIPTION
**Pull request by ahalberkamp**
In this pr the following was done:

1)Signing the artefacts before deploying to sonatype
2)Improve documentation.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,6 +3,8 @@ distributionPath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+
```
Comment (ahalberkamp) [inline gradle/wrapper/gradle-wrapper.properties:6]:
```suggestion
```

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -3,6 +3,8 @@ distributionPath=wrapper/dists
...
+
```
Comment (ahalberkamp) [inline gradle/wrapper/gradle-wrapper.properties]:
```suggestion
```

File: docs/BUILD.md [hunk 3]
```diff
@@ -5,6 +5,10 @@
 - [Release workflow](#release-workflow)
   - [Step 1: Pull request](#step-1-pull-request)
   - [Step 2: Release library in Github](#step-2-release-library-in-github)
+  - [Step 3: Publish to Maven central](#step-3-publish-to-maven-central)
```
Comment (ahmetoz) [inline docs/BUILD.md:8]: I wonder why we are not automating the release from "OSSRH" to "maven central" ? In that way, we don't need to document or do extra steps.
 
Check step 5: https://blog.sonatype.com/what-publishers-need-to-know-about-migrating-from-jcenter-/-bintray-to-the-central-repository

File: gradle-scripts/maven-publish.gradle [hunk 4]
```diff
@@ -53,3 +50,17 @@ publishing {
         }
     }
 }
+
+signing {
+    def signingKey = System.getenv("PGP_KEY")
```
Comment (ahmetoz) [inline gradle-scripts/maven-publish.gradle:55]: Can we recreate those when we lose the credentials? And what is the best way to share it in team ?

**Review by ahmetoz — CHANGES_REQUESTED** [review]

File: gradle-scripts/maven-publish.gradle (hunk 4 above)
Comment (ahalberkamp) [reply gradle-scripts/maven-publish.gradle:55]: Yes, the GPG-Key can easily be recreated using GPG and uploaded to github.

Comment (ahalberkamp) [reply gradle-scripts/maven-publish.gradle:55]: I think if we lost the key, we can just create a new one...-

File: docs/BUILD.md (hunk 3 above)
Comment (ahalberkamp) [reply docs/BUILD.md]: 217cbaa31

**Pull request merged**